src/lambda/commands/copyLambdaUrl.ts

```diff
@@ -0,0 +1,26 @@
+/*!
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DefaultLambdaClient, LambdaClient } from '../../shared/clients/lambdaClient'
+import { LambdaFunctionNode } from '../explorer/lambdaFunctionNode'
+import globals from '../../shared/extensionGlobals'
+import { copyToClipboard } from '../../shared/utilities/messages'
+import { addCodiconToString } from '../../shared/utilities/textUtilities'
+
+export const NO_URL_INFO_MSG = 'No URL for Lambda function.'
+
+export async function copyLambdaUrl(
+    node: LambdaFunctionNode,
+    client: LambdaClient = new DefaultLambdaClient(node.regionCode),
+    copyToClipboardFunc: (data: string, label?: string) => Promise<void> = copyToClipboard
+): Promise<void> {
+    const url = await client.getFunctionUrl(node.name)
+    if (!url) {
+        globals.window.showWarningMessage(NO_URL_INFO_MSG)
+        globals.window.setStatusBarMessage(addCodiconToString('circle-slash', NO_URL_INFO_MSG), 5000)
+        return
```

Comment:
Not sure if my [comment here](https://github.com/aws/aws-toolkit-vscode/pull/3036#discussion_r1042467640) changes the rationale to throw an error?

If no URL exists then we can direct them to the page on how to create the URL (I've updated the message to include the link to the docs as you mentioned before)

Reply:
Given that the lambda URL is opt-in, I don't think we can necessarily say that if there isn't a function URL it's an error (with the current code). I would have expected an info message saying something like "Function URL is not set-up: heres some instructions" (obviously in better wording).

I think in part, the issue is that it's not easy to ahead of time check enablement of the `Copy URL` right click option using the lambda client. We can technically list every lambda in one call, but checking if the lambda has a function URL requires another API call for every single lambda in the list (technically feasible, but I don't know enough about the AWS API to know if its costly or not). If we had a way to ahead of time check enablement of the node (if the node has a function URL), then, if we clicked a node and it didn't end up having a function url it would end up being an error. Without that implementation we can't really say this current state is an error.